internal/locate/region_cache.go

```diff
@@ -2078,7 +2105,8 @@ func (c *RegionCache) loadRegionByID(bo *retry.Backoffer, regionID uint64) (*Reg
 	}
 }
 
-// TODO(youjiali1995): for optimizing BatchLoadRegionsWithKeyRange, not used now.
+// For optimizing BatchLocateKeyRanges, scanRegionsFromCache scans at most `limit` regions from cache.
+// The first region's startKey is `startKey`, all regions returned are continuous and have no hole.
```

Comment:
maybe you need to emphasize that it is the caller's responsibility to make sure that `startKey` is a node in the B-tree,  otherwise, the `startKey` will not be included in the return regions?